### ISSUE TITLE
502 page not working

### ISSUE THREAD

**Issue by Sloth-on-meth** [linuxserver/docker-swag#180]
[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

------------------------------

## Expected Behavior
502.html should be used when a 502 error occurs

## Current Behavior
default NGINX 502 page shows.

## Steps to Reproduce
1.Cause a 502
2. visit the url
3.
4.

**Comment by drizuid** [comment]
confirmed by spad and I. the redirect as is ends up doing something like this
`upstream: "http://192.168.128.9:7878/config/www/502.html"`
when going to a site via proxy

**Comment by drizuid** [comment]
source of initial convo: https://discord.com/channels/354974912613449730/805754854982025246/907626669840404510

**Comment by drizuid** [comment]
the issue is that while the https://github.com/linuxserver/docker-swag/blob/00fde508251a69a04e0f877e19a1d2857924f5b5/root/defaults/default#L3 will work for the default, it will NOT work for any proxied site. the issue is that the proxy_pass line will parse and append the 502 path to that, so you get something like 
`upstream: "http://192.168.128.9:7878/config/www/502.html"`
the potential fix is to add another include file with contents like
```
# config for custom error files

    error_page 502 /502.html;
    location /502.html {
        root /config/www;
    }
```
and then include that into the proxy configs as such
```
# some random proxy config
...
    # enable for Authelia
    #include /config/nginx/authelia-server.conf;

    # enable for custom errors
    include /config/nginx/errors.conf;
...
```

**Comment by Sloth-on-meth** [comment]
any updates on this?

**Comment by drizuid** [comment]
> any updates on this?

I provided the fix you can use for now above; we will work a more permanent fix in at some later date

**Comment by Sloth-on-meth** [comment]
Issue still ongoing - bumping to prevent closure. Workaround in place

**Comment by drizuid** [comment]
> Issue still ongoing - bumping to prevent closure. Workaround in place

do not bump

**Comment by nemchik** [comment]
A fix is included in the WIP changes here https://github.com/linuxserver/docker-swag/pull/169/files#diff-25a343f59b2f053012602bec6f5916d2115b8403e686a4bd3e0e81127c55ac28

**Comment by HannesJo0139** [comment]
For me it is only working when I add the `error_page 502` or the `include /config/nginx/502.conf` statement to the location block. Adding it to the server block still leads to default error page. Anyone an idea whats could cause that?

**Comment by HannesJo0139** [comment]
> For me it is only working when I add the `error_page 502 ...` or the `include /config/nginx/502.conf` statement to the location block. Adding it to the server block still leads to default error page. Anyone an idea whats could cause that?

The workaround with error_page directive in server block only works as long as Authelia is not enabled. Using Authelia error_page must be in location block. Not sure why it is.

Also in order to use custom 404 error page one has to set
```
proxy_intercept_errors on;
fastcgi_intercept_errors on;
```
Otherwise eg. proxied heimdall app will just show  "File not found" on unknown .php files.

**Comment by nemchik** [comment]
We attempted a number of ways to resolve this and opted to remove the included custom 502 page in #169

We may re-address this in the future and add something back in.
